### ISSUE TITLE
Peaks and shape: display information

### ISSUE THREAD

**Issue by lpatiny** [cheminfo/nmrium#1669]
We need to be able to store the shape of the peak and optimize the shape as well.

The current structure of a peak is:
```json
  {
    "id": "1SMdLJ6q",
    "originalX": 3.8314013767193043,
    "x": 3.8314013767193043,
    "y": 67533497,
    "width": 0.9138402540488912
  }
```

It should be refactor to:
```json
  {
    "id": "1SMdLJ6q",
    "originalX": 3.8314013767193043,
    "x": 3.8314013767193043,
    "y": 67533497,
    "shape": {
      "kind": "gaussian",
      "fwhm": 0.9138402540488912
    }
  }
```

@jobo322 What do you think ? I guess the auto peak picking algorithm should also change in nmr-processing.

- [x] create a migration script in NMRium
- [x] change nmr-processing to generate the peaks with the right structure
- [ ] update NMRium peak panel to display those values (in the preferences)

**Comment by jobo322** [comment]
I think the last structure is more general and similar with the output of ml-spectra-fitting and/or ml-gsd/optimizePeaks

**Comment by jobo322** [comment]
it is also related with this https://github.com/mljs/spectra-fitting/issues/93

**Comment by lpatiny** [comment]
Finally we will keep width. Using our definition `width` is the width of the peak between the inflection points (obtained by GSD) while `fwhm` is the width half height.

```
  {
    "id": "1SMdLJ6q",
    "originalX": 3.8314013767193043,
    "x": 3.8314013767193043,
    "y": 67533497,
    "width": 0.95,
    "shape": {
      "kind": "gaussian",
      "fwhm": 0.9138402540488912
    }
  }
```

This means that we only need to add a new property and we don't need to have a migration script.

**Comment by lpatiny** [comment]
From preferences we should be able to display columns:
- shape.kind
- shape.fwhm
- shape.mu (only available if kind = pseudovoigt

Be able to edit the kind using a dropdown select. 3 values:
- gaussian
- lorentzian
- pseudovoigt

**Comment by lpatiny** [comment]
@hamed-musallam could you allow in the peak preferences to add those 3 columns (shape.kind, shape.fwhm and shape.mu)

**Comment by jobo322** [comment]
In version 8.5.1 of nmr-processing we export the right structure for it.

**Comment by hamed-musallam** [comment]
closed by #1683

**Comment by hamed-musallam** [comment]
closed by #1683
